apps/wallet-ui/src/components/ConnectButton.tsx

```diff
@@ -0,0 +1,42 @@
+"use client";
+import {
+  generatePayload,
+  isLoggedIn,
+  login,
+  logout,
+} from "@/app/[ecosystem]/login/auth";
+import { client } from "@/lib/client";
+import { useTheme } from "next-themes";
+import {
+  arbitrum,
+  base,
+  blast,
+  mainnet,
+  optimism,
+  zkSync,
+} from "thirdweb/chains";
+import { ConnectButton as ThirdwebConnectButton } from "thirdweb/react";
+import { ecosystemWallet } from "thirdweb/wallets";
+
+export default function ConnectButton({
+  ecosystem,
+}: { ecosystem: `ecosystem.${string}` }) {
+  const { theme } = useTheme();
+
+  return (
+    <ThirdwebConnectButton
+      chains={[mainnet, base, optimism, arbitrum, blast, zkSync]}
+      wallets={[ecosystemWallet(ecosystem)]}
+      client={client}
+      theme={theme as "light" | "dark"}
+      auth={{
+        getLoginPayload: async (params) => generatePayload(params),
+        doLogin: async (params) => {
+          await login(params);
+        },
+        isLoggedIn,
+        doLogout: logout,
+      }}
```

Comment:
does this truly need auth? or is auth enforced on the page level anyways? (pages need to be authed and redirect to `/login` if not)

Reply:
For the only page its currently on, no. I added it in the case of a non-gated page that this button might appear on.